test/basic.jl

```diff
@@ -22,6 +22,16 @@ using InteractiveUtils
     @test @jit(sum(a)) ≈ r_res
 end
 
+@testset "Julia Compilation cache" begin
+    x = @compile -(Reactant.to_rarray(ones(2)))
+    y = @compile -(Reactant.to_rarray(ones(2)))
+
+    @test typeof(x) == typeof(y)
+    # TODO, currently x and y are not equal as x.exec != y.exec
+    # as the executable we generate is itself not cached
+    # (which clearly we should do to improve jit time)
```

Comment:
following the comment above, what if we add this test...

```suggestion
    x = @compile -(Reactant.to_rarray(ones(2)))
    y = @compile -(Reactant.to_rarray(ones(2)))
    z = @compile -(Reactant.to_rarray(ones(3)))
    
    @test x(Reactant.to_rarray(ones(2))) == -ones(2)
    @test y(Reactant.to_rarray(ones(2))) == -ones(2)
    @test z(Reactant.to_rarray(ones(3))) == -ones(3)

    @test typeof(x) == typeof(y)
    # TODO, currently x and y are not equal as x.exec != y.exec
    # as the executable we generate is itself not cached
    # (which clearly we should do to improve jit time)
```

Reply:
so that is independent of this pr [just that re compiles produce equivalent fns], so I presume tested elsewhere?